scripts/mount-admin-tools-bucket.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+# MIT License
+#
+# (C) Copyright [2024] Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+# Mount admin-tools S3 bucket
+s3_bucket="admin-tools"
+s3fs_mount_dir="/var/lib/admin-tools"
+s3_user="admin-tools"
+
+s3fs_cache_dir=/var/lib/s3fs_cache
+if [ -d ${s3fs_cache_dir} ]; then
+  s3fs_opts="use_path_request_style,use_cache=${s3fs_cache_dir},check_cache_dir_exist,use_xattr"
+else
+  s3fs_opts="use_path_request_style,use_xattr"
+fi
+
+echo "Configuring for ${s3_bucket} S3 bucket at ${s3fs_mount_dir} for ${s3_user} S3 user"
+
+mkdir -p ${s3fs_mount_dir}
+
+pwd_file=/root/.${s3_user}.s3fs
+access_key=$(kubectl get secret ${s3_user}-s3-credentials -o json | jq -r '.data.access_key' | base64 -d)
+secret_key=$(kubectl get secret ${s3_user}-s3-credentials -o json | jq -r '.data.secret_key' | base64 -d)
+s3_endpoint=$(kubectl get secret ${s3_user}-s3-credentials -o json | jq -r '.data.http_s3_endpoint' | base64 -d)
+
+echo "${access_key}:${secret_key}" > ${pwd_file}
+chmod 600 ${pwd_file}
+
+echo "Mounting bucket: ${s3_bucket} at ${s3fs_mount_dir}"
+s3fs ${s3_bucket} ${s3fs_mount_dir} -o passwd_file=${pwd_file},url=${s3_endpoint},${s3fs_opts}
+
+echo "Adding fstab entry for ${s3_bucket} S3 bucket at ${s3fs_mount_dir} for ${s3_user} S3 user"
+if [[ -z $(cat /etc/fstab | grep "admin-tools" | grep "fuse.s3fs") ]]; then
```

Comment:
Pro-tips:
1. `grep` instead of `cat | grep`.
2. `grep -q` sets the exit code, since you use multiple `greps` it's only needed on the final `grep`
```suggestion
if grep "${s3_bucket}" /etc/fstab | grep -q "fuse.s3fs"; then
```

In scripts where you do not have `set -o pipefail`, this reduces the need for a pipe. In  general, it's extraneous.
Same for `cat | jq`, `cat | yq`.

As for the `-q` flag, this removes the need to use any brackets or check ambiguous output. `grep -q` will simply set `$?` to `0` or `1`.